src/interpreter/ast.rs

```diff
@@ -1,11 +1,13 @@
 use std::fmt;
 use std::sync::Arc;
 
+use serde::{Deserialize, Serialize};
```

Comment:
Just a matter of style, but elsewhere we don't import the derives, and just derive `serde::Serialize`. Seems a bit cleaner to me.